flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCUpsertOutputFormat.java

```diff
@@ -286,6 +286,7 @@ public Builder setMaxRetryTimes(int maxRetryTimes) {
 		public JDBCUpsertOutputFormat build() {
 			checkNotNull(options, "No options supplied.");
 			checkNotNull(fieldNames, "No fieldNames supplied.");
+			checkNotNull(fieldTypes, "No fieldTypes supplied.");
```

Comment:
fieldTypes should can not supplied?
If not supplied, go a default way?

Reply:
As @wuchong mentioned, it is error-prone if we don't set field types. If field types are not set, the `upload.setObject` in `setRecordToStatement` will try to directly set a `java.time.LocalDate` into a `DATE` type field.